### PR TITLE
[ADD] Support KFAC if the network has inplace activations

### LINKED ISSUE #58: Bug: Networks with in-place operations not supported in KFAC
This is similar to https://github.com/f-dangel/singd/issues/56 and caused by the implementation of computing the gradient covariance via `full_backward_hook`s. The fix is to replace the module backward hook with a tensor backward hook installed on the output of a layer.

### PR DESCRIPTION
**Pull request by f-dangel**
Resolves #58.

### CODE REVIEW

File: test/test_kfac.py [hunk 1]
```diff
@@ -103,3 +105,49 @@ def test_kfac_ef_one_datum(
...
+
+    kfac = KFACLinearOperator(model, loss_func, params, data, mc_samples=2_000)
+    kfac_mat = kfac @ eye(kfac.shape[1])
+
+    atol = {"sum": 5e-1, "mean": 2e-3}[loss_func.reduction]
+    rtol = {"sum": 2e-2, "mean": 2e-2}[loss_func.reduction]
+
+    report_nonclose(ggn, kfac_mat, rtol=rtol, atol=atol)
+
+    # 2) Compare GGN (inplace=True) and GGN (inplace=False)
```
Comment (runame) [inline test/test_kfac.py:142]: Shouldn't we here compare "GGN (inplace=True) against KFAC (inplace=False)" or equivalently "KFAC (inplace=True) against KFAC (inplace=False)".

**Review by runame — CHANGES_REQUESTED** [review]
Can't we also apply the same refactor as in SINGD, where we just use a single forward hook that computes the input covariance and installs the tensor hook on the module output?

**Comment by f-dangel** [comment]
One point against merging these hooks is that we may want to re-cycle the hook for computing the activation variance (e.g. FOOF).

**Comment by runame** [comment]
Ok let's stick to separate hooks for now then.

**Review by runame — APPROVED** [review]

**Pull request merged**